src/views/Login.vue

```diff
@@ -1,12 +1,22 @@
 <template>
   <div class="page-wrapper">
     <h1 class="login-page-title">Login page</h1>
+
     <!-- Loader -->
-    <div v-if="user === undefined">loading ...</div>
+    <div class="loader" v-show="user === undefined">loading ...</div>
+
+    <!-- Offline instruction -->
+    <div class="offline-instruction" v-show="!networkOnLine">
+      Please check your connection, login feature is not available offline.
+    </div>
 
     <p v-if="loginError">{{ loginError }}</p>
     <!-- Auth UI -->
-    <div class="login-btn" v-show="user !== undefined && !user" @click="login">
+    <div
+      class="login-btn"
+      v-show="user !== undefined && !user && networkOnLine"
```

Comment:
should we put this condition in a computed property `showLoginButton` of the component to have a clearer template ?

Reply:
Good idea !